api/src/main/java/jakarta/persistence/TypedQuery.java

```diff
@@ -314,8 +358,42 @@ TypedQuery<X> setParameter(String name, Date value,
      *         correspond to a positional parameter of the
      *         query or if the argument is of incorrect type
      */
+    @Override
     TypedQuery<X> setParameter(int position, Object value);
 
+
+    /**
+     * Bind an argument value to a named parameter, explicitly
+     * specifying the parameter type. This is most useful when
+     * the argument might be null.
+     * @param position  position
+     * @param value  parameter value
+     * @param type  a class object representing the parameter type
+     * @return the same query instance
+     * @throws IllegalArgumentException if position does not
+     *         correspond to a positional parameter of the
+     *         query or if the argument is of incorrect type
+     * @since 4.0
+     */
+    @Override
+    <P> TypedQuery<X> setParameter(int position, P value, Class<P> type);
+
+    /**
+     * Bind an argument value to a named parameter, explicitly
+     * specifying the parameter type. This is most useful when
+     * the binding is affected by an attribute converter.
```

Comment:
Copying an example to TypedQuery, so that we also show the usage on the interface that is expected to be used for SELECT queries,
```suggestion
     * the binding is affected by an attribute converter.
     * {@snippet :
     * var amount = MonetaryAmount.of(priceLimit, currency);
     * var affordableBooks =
     *     em.createQuery("from Book where price < ?1", Book.class)
     *         .setParameter(1, amount, Book_.price.getType())
     *         .getResultList();
     * }
```